### DISCUSSION TITLE
OpenTelemetry Middleware

### DISCUSSION THREAD

**Discussion by hdost** [http4s/http4s#5799] [Ideas]
Is there any interest in introducing OpenTelemetry Middleware?

**Comment by armanbilge** [comment]
Is this something that can be accomplished with https://github.com/tpolecat/natchez and https://github.com/tpolecat/natchez-http4s?

**Reply by bplommer** [reply]
See also https://github.com/trace4cats/trace4cats - it's based on Natchez but it has a lot more integration modules.

**Comment by armanbilge** [comment]
This is now available in https://github.com/davenverse/natchez-http4s-otel.

**Comment by rossabaker** [comment]
I'm also working on https://github.com/rossabaker/otel4s.  I am working on metrics before tracing, and [editorial] I think metrics and tracing are better done in backends than middleware.  But if someone wants to give a metrics middleware a go based on that... it exists.
